http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/ChannelPipelineInitializer.java

```diff
@@ -167,11 +170,52 @@ private void configureHttp11(Channel ch, ChannelPipeline pipeline) {
         ch.attr(PROTOCOL_FUTURE).get().complete(Protocol.HTTP1_1);
     }
 
+    private void configureAlpnH2(ChannelPipeline pipeline) {
+        pipeline.addLast(new ApplicationProtocolNegotiationHandler("") {
+            @Override
+            protected void configurePipeline(ChannelHandlerContext ctx, String protocol) {
+                if (ApplicationProtocolNames.HTTP_2.equals(protocol)) {
```

Comment:
Should we handle the case where the returned protocol is not h2? 
Can you point to me the logic where we handle the case where alpn is not supported on the server side?

Reply:
Updated to throw UnsupportedOperationException